bigquery/monitor-tables-for-errors.sql

```diff
@@ -15,27 +15,56 @@ IF
     IF
       (table LIKE "%school"
         AND row_count <28000,
-        "Table appears incomplete", #check whether there are at least this many schools in the schools table
+        "Table appears incomplete",
+        #check whether there are at least this many schools in the schools table
       IF
-        ((table ="users"
-            AND (
-            SELECT
-              MAX(update_datetime)
-            FROM
-              `teacher-vacancy-service.production_dataset.users` ) < TIMESTAMP_SUB(CURRENT_TIMESTAMP(),INTERVAL 3 DAY)) #check the last updated date for any user's data from DSI - produce error if more than three days old (i.e. if longer ago than the last working day)
-          OR (table LIKE "feb20_%"
-            AND (
-            SELECT
-              MAX(updated_at)
-            FROM
-              `teacher-vacancy-service.production_dataset.feb20_audit_data` ) < TIMESTAMP_SUB(CURRENT_TIMESTAMP(),INTERVAL 3 DAY)) OR (table IN("alert_run","audit_data","detailed_school_type","general_feedback","leadership","pay_scale","region","school","school_type","subject","subscription","transaction_auditor","user","vacancy","vacancy_publish_feedback") #check the last entry in the audit log in the feb20_ tables from our database - error if this was more than 3 days ago
-            AND (
-            SELECT
-              MAX(PARSE_DATETIME("%e %B %E4Y %R",string_field_3))
-            FROM
-              `teacher-vacancy-service.production_dataset.audit_data` ) < DATETIME_SUB(CURRENT_DATETIME(),INTERVAL 3 DAY)), #check the last entry in the audit log in the legacy tables from our database - error if this was more than 3 days ago
-          "Latest date a row was updated in a frequently updated table from this source was at least 3 days ago",
-          "OK")))) AS status
+        (table = "dsi_users"
+          AND row_count <25000,
```

Comment:
Are these arbitrary, and can we define them?

Reply:
@cpjmcquillan I looked into this before, and there isn't a straightforward way of doing this in BigQuery SQL - although I agree it would be nice to have! I'll add a comment that explains what this does though.